src/Tasks/Backup/Gzip.php

```diff
@@ -0,0 +1,65 @@
+<?php
+
+namespace Spatie\Backup\Tasks\Backup;
+
+class Gzip
+{
+    /** @var string */
+    public $originalFilePath;
+
+    /** @var string */
+    public $filePath;
+
+    /** @var bool */
+    public $failed;
+
+    public function __construct(string $filePath)
+    {
+        $this->originalFilePath = $filePath;
+
+        $this->filePath = $this->compress();
+
+        $this->failed = $this->originalFilePath == $this->filePath;
+    }
+
+    /**
+     * @return string
+     */
+    protected function compress()
+    {
+        $gzipPath = $this->originalFilePath.'.gz';
+
+        if ($gzipOut = gzopen($gzipPath, 'w9')) {
+            if ($gzipIn = fopen($this->originalFilePath, 'rb')) {
+                while (! feof($gzipIn)) {
+                    gzwrite($gzipOut, fread($gzipIn, 1024 * 512));
+                }
+                fclose($gzipIn);
+            } else {
```

Comment:
I should have also mentioned that I really dislike `else` statements. Could you refactor this code to avoid them entirely?